File: frontend/src/services/scheduleService.ts
import api from "./api";
import type { Schedule } from "../types/schedule";


export const getSchedules = async (): Promise<Schedule[]> => {

    const response = await api.get("/Schedule");

    return response.data;

};


export const createSchedule = async (
    schedule: Schedule
) => {

    const response = await api.post(
        "/Schedule",
        schedule
    );

    return response.data;

};


export const deleteSchedule = async (
    id: number
) => {

    await api.delete(
        `/Schedule/${id}`
    );

};

File: backend/BookingService.API/Services/ScheduleService.cs
using BookingService.API.Models;
using BookingService.API.Repositories;

namespace BookingService.API.Services;

public class ScheduleService
{
    private readonly ScheduleRepository repository;

    public ScheduleService(ScheduleRepository repository)
    {
        this.repository = repository;
    }

    public async Task<List<Schedule>> GetSchedules()
    {
        return await repository.GetSchedules();
    }

    public async Task<Schedule?> GetSchedule(int id)
    {
        return await repository.GetSchedule(id);
    }

    public async Task<Schedule> CreateSchedule(Schedule schedule)
    {
        return await repository.CreateSchedule(schedule);
    }

    public async Task UpdateSchedule(Schedule schedule)
    {
        await repository.UpdateSchedule(schedule);
    }

    public async Task DeleteSchedule(int id)
    {
        await repository.DeleteSchedule(id);
    }
}